plugin/src/main/java/com/denizenscript/denizen/utilities/entity/EntityMetadataCommandHelper.java

```diff
@@ -0,0 +1,126 @@
+            return;
+        }
+        NetworkInterceptHelper.enable();
+        boolean wasEntityAdded = !packetOverrides.containsKey(target.getUUID());
+        Map<UUID, Boolean> playerMap = packetOverrides.computeIfAbsent(target.getUUID(), k -> new HashMap<>());
+        for (PlayerTag player : players) {
+            boolean state = stateSupplier.test(player);
+            Boolean oldState = playerMap.put(player.getUUID(), state);
+            if ((wasEntityAdded || oldState == null || oldState != state) && player.isOnline()) {
+                NMSHandler.packetHelper.sendEntityMetadataFlagsUpdate(player.getPlayerEntity(), target.getBukkitEntity());
+            }
+        }
+    }
+
+    public Boolean getState(Entity entity, UUID player, boolean fakeOnly) {
+        if (entity == null) {
+            return null;
+        }
+        if (player != null) {
+            Map<UUID, Boolean> playerMap = packetOverrides.get(entity.getUniqueId());
+            if (playerMap != null && playerMap.containsKey(player)) {
+                return playerMap.get(player);
+            }
+        }
+        if (fakeOnly) {
+            return null;
+        }
+        return getter.test(entity);
+    }
+
+    public boolean noOverrides() {
+        return packetOverrides.isEmpty();
+    }
+
+    public enum Action {TRUE, FALSE, TOGGLE, RESET}
+
+    public void execute(ScriptEntry scriptEntry, List<EntityTag> targets, Action action, List<PlayerTag> forPlayers) {
+        if (targets == null) {
+            targets = Utilities.entryDefaultEntityList(scriptEntry, true);
+            if (targets == null) {
+                throw new InvalidArgumentsRuntimeException("Must specify valid targets.");
+            }
+        }
+        switch (action) {
+            case TRUE, FALSE -> {
+                boolean state = action == Action.TRUE;
+                if (forPlayers == null) {
+                    for (EntityTag target : targets) {
+                        setter.accept(target, state);
+                    }
+                }
+                else {
+                    for (EntityTag target : targets) {
+                        setForPlayers(forPlayers, target, player -> state);
+                    }
+                }
+            }
+            case TOGGLE -> {
+                if (forPlayers == null) {
+                    for (EntityTag target : targets) {
+                        setter.accept(target, !getState(target.getBukkitEntity(), null, false));
+                    }
+                }
+                else {
+                    for (EntityTag target : targets) {
+                        setForPlayers(forPlayers, target, player -> !getState(target.getBukkitEntity(), player.getUUID(), false));
+                    }
+                }
+            }
+            case RESET -> {
+                for (EntityTag target : targets) {
+                    Map<UUID, Boolean> playerMap = packetOverrides.get(target.getUUID());

```

Comment:
This should be `.remove` here (remove returns the value and removes in one op)

Reply:
Might be missing something, but I don't think that's valid? you can reset for specific players, E.g. if an entity has a fake state for 4 players, and you use `reset for:<2 of these players>`, the map should still be there and contain the 2 remaining players.

Reply:
Oh, true. I saw a bunch of removes and missed that there's conditions that don't lead to it.